Test/Unit/Context.php

```diff
@@ -0,0 +1,226 @@
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Hoa nor the names of its contributors may be
+ *       used to endorse or promote products derived from this software without
+ *       specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Math\Test\Unit;
+
+use Hoa\Core;
+use Hoa\Math as LUT;
+use Hoa\Math\Context as CUT;
+use Hoa\Test;
+
+/**
+ * Class \Hoa\Math\Test\Unit\Context.
+ *
+ * Test suite of the Hoa\Math\Context class.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+class Context extends Test\Unit\Suite
+{
+    public function case_context_has_no_predefined_variable()
+    {
+        $this
+            ->given($context = new CUT())
+            ->then
+                ->array(iterator_to_array($context->getVariables()))->isEmpty();
+    }
+
+    public function case_context_exception_when_getting_unknown_variable()
+    {
+        $this
+            ->given(
+                $name    = 'foo',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->getVariable($name);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownVariable')
+                    ->hasMessage('Variable ' . $name .' does not exist');
+    }
+
+    public function case_context_returns_variable_value()
+    {
+        $this
+            ->given(
+                $name     = 'foo',
+                $value    = 42,
+                $callable = function() use ($value) { return $value; },
+                $context  = new CUT()
+            )
+            ->when($context->addVariable($name, $callable))
+            ->then
+                ->variable($context->getVariable($name))->isEqualTo($value);
+    }
+
+    public function case_context_has_predefined_constants()
+    {
+        $this
+            ->given($context = new CUT())
+            ->then
+                ->array(iterator_to_array($context->getConstants()))
+                    ->isNotEmpty()
```

Comment:
What about `isEqualTo`? Please, strictly test the size of the array else, in order to test “regressions” when we will add new constants.